src/datasets/tasks/text_classification.py

```diff
@@ -23,30 +24,26 @@ def __get__(self, obj, objtype=None):
 
 @dataclass(frozen=True)
 class TextClassification(TaskTemplate):
-    task: ClassVar[str] = "text-classification"
+    # `task` is not a ClassVar since we want it to be part of the `asdict` output for JSON serialization
+    task: str = "text-classification"
```

Comment:
This fixed your issue when reloading the dataset @lewtun 
Indeed `task_template_from_dict` wasn't able to reload the template from json data since the "task" field was missing

Reply:
amazing, thanks for spotting and fixing this!